server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/DefaultHighlighter.java

```diff
@@ -34,31 +37,21 @@
 import java.io.IOException;
 import java.text.BreakIterator;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
-import java.util.Map;
 import java.util.function.Predicate;
 
 import static org.elasticsearch.lucene.search.uhighlight.CustomUnifiedHighlighter.MULTIVAL_SEP_CHAR;
 
-public class UnifiedHighlighter implements Highlighter {
+public class DefaultHighlighter implements Highlighter {
     @Override
     public boolean canHighlight(MappedFieldType fieldType) {
         return true;
     }
 
     @Override
     public HighlightField highlight(FieldHighlightContext fieldContext) throws IOException {
-        @SuppressWarnings("unchecked")
```

Comment:
With this change we don't reuse the per-field highlighters on subsequent hits.  I guess building the highlighter isn't a particularly expensive operation, but did you intend to remove this?

Reply:
Good catch, I removed it to check a failure and forgot to re-add it. I pushed https://github.com/elastic/elasticsearch/pull/96068/commits/9cd6e3ddcec77a8ea0fbf6c3c08efb46e9b0832d